tests/mock/test_column.py

```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from snowflake.snowpark.functions import col, when
+from snowflake.snowpark.row import Row
+
+
+def test_casewhen_with_non_zero_row_index(session):
+    df = session.create_dataframe([[1, 2], [3, 4]], schema=["a", "b"])
+    assert df.filter(col("a") > 1).select(
+        when(col("a").is_null(), 5).when(col("a") == 1, 6).otherwise(7).as_("a")
+    ).collect() == [Row(A=7)]
+
+
+def test_regexp_with_non_zero_row_index(session):
+    df = session.create_dataframe([["1", 2], ["3", 4]], schema=["a", "b"])
+    assert df.filter(col("b") > 2).select(
+        col("a").like("1").alias("res")
```

Comment:
shouldn't this be `col("a").regexp().alias("res")` if we are testing regexp?

Reply:
I revisited the PR and it is to fix `col().like()`. a nit here would be to rename the test to `test_like_with_non_zero_row_index`. Just a follow-up, do you know if regexp has the same issue and if your PR will address that issue too?

Reply:
Nice catch! `regexp` does not have this issue.